git@bitbucket.org auth failed with valid ssh keys

Java: 17 LTS
Maven: 3.8.7

From the terminal
`git clone git@bitbucket.org:company/repo1.git`
`git clone git@bitbucket.org:company/repo2.git`
works fine

However:
`./run.sh git@bitbucket.org:company/repo1.git:aa git@bitbucket.org:company/repo2.git:bb`

```
[ERROR] Failed to execute goal org.codehaus.mojo:exec-maven-plugin:1.4.0:java (default-cli) on project git-merge-repos: An exception occured while executing the Java class. null: InvocationTargetException: Invalid remote: git@bitbucket.org: git@bitbucket.org: Auth fail. -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
```

I have valid ssh keys in ~/.ssh

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        java: [8, 17]
+        java: [11, 17]
     steps:
       - name: Checkout
         uses: actions/checkout@v3

--- a/README.md
+++ b/README.md
@@ -75,7 +75,7 @@ we already did that using `git filter-repo`.
 Dependencies
 ------------
 
-You will need to have at least Java 8 installed. Note that it's also tested against newer versions such as 17.
+You need to have at least Java 11 installed. Note that it's also tested against newer versions such as 17.
 [Maven][maven], [JGit][jgit] and other dependencies will be automatically downloaded when running the program.
 
 Other Solutions

--- a/pom.xml
+++ b/pom.xml
@@ -35,12 +35,12 @@
         <dependency>
             <groupId>org.eclipse.jgit</groupId>
             <artifactId>org.eclipse.jgit</artifactId>
-            <version>4.0.1.201506240215-r</version>
+            <version>6.6.0.202305301015-r</version>
         </dependency>
         <dependency>
             <groupId>org.slf4j</groupId>
             <artifactId>slf4j-jdk14</artifactId>
-            <version>1.7.2</version>
+            <version>1.7.36</version>
         </dependency>
     </dependencies>
 </project>

--- a/pom.xml
+++ b/pom.xml
@@ -21,8 +21,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.11.0</version>
                 <configuration>
-                    <source>1.7</source>
-                    <target>1.7</target>
+                    <source>11</source>
+                    <target>11</target>
                 </configuration>
             </plugin>
             <plugin>

--- a/src/main/java/org/nibor/git_merge_repos/MergedRef.java
+++ b/src/main/java/org/nibor/git_merge_repos/MergedRef.java
@@ -45,8 +45,7 @@ public class MergedRef {
 			appendRepositoryNames(messageBuilder, configsWithoutRef);
 		}
 		messageBuilder.append("\n");
-		String message = messageBuilder.toString();
-		return message;
+		return messageBuilder.toString();
 	}
 
 	private static void appendRepositoryNames(StringBuilder builder,

--- a/src/main/java/org/nibor/git_merge_repos/RepoMerger.java
+++ b/src/main/java/org/nibor/git_merge_repos/RepoMerger.java
@@ -99,10 +99,10 @@ public class RepoMerger {
 		try (RevWalk revWalk = new RevWalk(repository)) {
 			Collection<Ref> refs = new ArrayList<>();
 			RefDatabase refDatabase = repository.getRefDatabase();
-			Map<String, Ref> originalBranches = refDatabase.getRefs("refs/heads/original/");
-			Map<String, Ref> originalTags = refDatabase.getRefs("refs/tags/original/");
-			refs.addAll(originalBranches.values());
-			refs.addAll(originalTags.values());
+			List<Ref> originalBranches = refDatabase.getRefsByPrefix("refs/heads/original/");
+			List<Ref> originalTags = refDatabase.getRefsByPrefix("refs/tags/original/");
+			refs.addAll(originalBranches);
+			refs.addAll(originalTags);
 			for (Ref originalRef : refs) {
 				RefUpdate refUpdate = repository.updateRef(originalRef.getName());
 				refUpdate.setForceUpdate(true);
@@ -113,7 +113,7 @@ public class RepoMerger {
 
 	private void resetToBranch() throws IOException, GitAPIException {
 		for (String name : Arrays.asList("main", "master")) {
-			Ref branch = repository.getRef(Constants.R_HEADS + name);
+			Ref branch = repository.exactRef(Constants.R_HEADS + name);
 			if (branch != null) {
 				Git git = new Git(repository);
 				git.reset().setMode(ResetType.HARD).setRef(branch.getName()).call();
@@ -237,10 +237,15 @@ public class RepoMerger {
 	}
 
 	private Collection<String> getRefSet(String prefix) throws IOException {
-		Map<String, Ref> refs = repository.getRefDatabase().getRefs(prefix);
+		List<Ref> refs = repository.getRefDatabase().getRefsByPrefix(prefix);
 		TreeSet<String> result = new TreeSet<>();
-		for (String refName : refs.keySet()) {
-			String branch = refName.split("/", 2)[1];
+		for (Ref ref : refs) {
+			// full: refs/heads/original/repo-one/main
+			String full = ref.getName();
+			// unprefixed: repo-one/main
+			String unprefixed = full.substring(prefix.length());
+			// branch name: main
+			String branch = unprefixed.split("/", 2)[1];
 			result.add(branch);
 		}
 		return result;

--- a/src/main/java/org/nibor/git_merge_repos/SubtreeMerger.java
+++ b/src/main/java/org/nibor/git_merge_repos/SubtreeMerger.java
@@ -1,6 +1,7 @@
 package org.nibor.git_merge_repos;
 
 import java.io.IOException;
+import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
@@ -11,9 +12,6 @@ import java.util.Map;
 import org.eclipse.jgit.dircache.DirCache;
 import org.eclipse.jgit.dircache.DirCacheBuilder;
 import org.eclipse.jgit.dircache.DirCacheEntry;
-import org.eclipse.jgit.errors.CorruptObjectException;
-import org.eclipse.jgit.errors.IncorrectObjectTypeException;
-import org.eclipse.jgit.errors.MissingObjectException;
 import org.eclipse.jgit.lib.CommitBuilder;
 import org.eclipse.jgit.lib.ObjectId;
 import org.eclipse.jgit.lib.ObjectInserter;
@@ -23,7 +21,6 @@ import org.eclipse.jgit.revwalk.RevCommit;
 import org.eclipse.jgit.treewalk.AbstractTreeIterator;
 import org.eclipse.jgit.treewalk.CanonicalTreeParser;
 import org.eclipse.jgit.treewalk.TreeWalk;
-import org.eclipse.jgit.util.RawParseUtils;
 
 /**
  * Merges the passed commit trees into one tree, adjusting directory structure
@@ -107,7 +104,7 @@ public class SubtreeMerger {
 			if (".".equals(directory)) {
 				treeWalk.addTree(parentCommit.getTree());
 			} else {
-				byte[] prefix = directory.getBytes(RawParseUtils.UTF8_CHARSET);
+				byte[] prefix = directory.getBytes(StandardCharsets.UTF_8);
 				CanonicalTreeParser treeParser = new CanonicalTreeParser(prefix,
 						treeWalk.getObjectReader(), parentCommit.getTree());
 				treeWalk.addTree(treeParser);

